src/main/java/org/mockito/internal/stubbing/answers/ReturnsArgumentAt.java

```diff
@@ -51,24 +51,26 @@ public ReturnsArgumentAt(int wantedArgumentPosition) {
     }
 
     @Override
-    public Object answer(InvocationOnMock invocation) throws Throwable {
-        int argumentPosition = inferWantedArgumentPosition(invocation);
-        validateIndexWithinInvocationRange(invocation, argumentPosition);
+    public Object answer(InvocationOnMock invocationOnMock) throws Throwable {
+        Invocation invocation = (Invocation) invocationOnMock;
 
-        if (wantedArgIndexIsVarargAndSameTypeAsReturnType(
-                invocation.getMethod(), argumentPosition)) {
+        if (wantedArgIndexIsVarargAndSameTypeAsReturnType(invocation)) {
             // answer raw vararg array argument
-            return ((Invocation) invocation).getRawArguments()[argumentPosition];
```

Comment:
Note the existing cast of `InvocationOnMock` to `Invocation` to access the `getRawArguments` method. 

What do you think to exposing `getRawArguments` on the `InvocationOnMock` interface, and thereby removing the need for  lots of casts to get to the `Invocation`?

Reply:
If I remember correctly, we do these casts in quite a few place. If that's indeed the case, then yes let's introduce that method. If this is the only place, let's keep the cast here.

Reply:
Agreed.  When making it public would you also want me to make the method return a copy of any array parameters, so that we're not exposing mutable internal state?

Reply:
Yes makes sense.

Reply:
Argh, I've noticed that we don't make copies when returning arrays from `getArgument`. So I'll stay consistent and not create a copy for `getRawArgument`.

Reply:
Actually, I notice that `getArguments` does not do an array copy, so I'll stick with the same pattern when exposing `getRawArguments`.